spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAddFilesProcedure.java

```diff
@@ -103,6 +103,35 @@ public void addDataUnpartitioned() {
         sql("SELECT * FROM %s ORDER BY id", tableName));
   }
 
+  @Test
+  public void deleteAndAddBackAllowed() {
```

Comment:
wondering if there will be any additional scenarios we want to test for, especially how this would behave wrt merge-on-read as paths will be duplicates

Reply:
I think this is a good callout to verify the add-files behaviour for v2 in general, which I dont see any coverage in existing tests. 

Focus on this particular PR, I think no matter v1 or v2, we shall not rely duplicate check based on file path only, the file path with deleted status will go away on next snapshot, so the current workaround before this fix is simply force a new snapshot between DELETE and add-files procedure call.